### DISCUSSION TITLE
How to bulk add links from lists, html files, csv etc?

### DISCUSSION THREAD

**Discussion by joaosimuai** [KnowledgeCanvas/knowledge#81] [Q&A] [answered]
Hi, I'm a data accumulator and was trying to make sense of it by trying Data Cleaning and Wrangling tools for the first time, for the purpose of organizing a big list of URLs. Then I found your tool and it seems great. Thank you! I've begun to use it just now. I've come with this question because I couldn't find how anywhere. Files exported from other browsers and apps are only imported as items, without contact with their content it seems. Even if I organize a proper URL-only list from them, the only option I see is to add them individually. Is that so or am I doing something wrong?

**Comment by RobRoyce** [comment, marked as answer]
Thanks for the feedback!

You are correct: currently, the tool only saves links and various metadata about imported items. I have been planning to support full content extraction, but need to rework the data storage mechanism for it to be viable.

As for importing multiple items at a time, that is currently limited to files (drag-and-drop, multi-select in file browser, etc.). I think the quickest way to add links would be drag-and-drop one at a time (drag the link into the window).

That said, I think we should change the URL import to allow for multiple items. It would be cool to copy/paste a newline-delimited list and parse each link as an individual link. Also drag/drop for multiple links could be viable as well.

I'll add this to the roadmap. Thanks for the suggestion!
